### DISCUSSION TITLE
Editor doesn't save created maps anywhere Linux

### DISCUSSION THREAD

**Discussion by LargerManZ** [ppy/osu#18869] [Q&A] [answered]
### Type

Game behaviour

### Bug description

I created a map using the editor, I accidentally had it set to taiko when making it, so I created a new difficulty and gave it a name and an audio file and filled out some of the boxes for information, like the artist. I went through and timed it and then placed some circles and then saved it using the save button in the File menu. I then closed it and it asked me if I wanted to save, to which I said yes. (couldn't be too careful) And low and behold. Looking for my song in my songs list, it isn't there. And there is no way to open up other projects in the editor menus so I don't even know if it's in the files whatsoever. 

_(make a way to delete difficulties inside the setup menu pls)_ 

No disclaimer that the beatmap creation is completely dysfunctional or doesn't save maps, so I'm not sure if it just isn't flushed out yet or if this is a linux unique bug or something.

### Version

Lazer 2022.621.0

### Logs

[database.log](https://github.com/ppy/osu/files/8987872/database.log)
[network.log](https://github.com/ppy/osu/files/8987873/network.log)
[performance.log](https://github.com/ppy/osu/files/8987874/performance.log)
[runtime.log](https://github.com/ppy/osu/files/8987875/runtime.log)

**Comment by smoogipoo** [comment]
Do you have the correct ruleset selected in the top-left of the toolbar? And/or perhaps this setting turned off in the top right of song select?:
![image](https://user-images.githubusercontent.com/1329837/175855526-7e81cdf1-718f-432a-8e3a-654c12920f61.png)

**Comment by LargerManZ** [comment]
yes I do have [show converted] on, and the song [Edit: Map] I made doesn't show up regardless of the gamemode I have it set to.

**Comment by peppy** [comment]
From your logs:

```csharp
2022-06-26 23:54:01 [error]: Camellia · Nanahira - Artificial Intelligence (BreadKnightRG) already has a difficulty with the name of 'Automaton'.
2022-06-26 23:54:01 [error]: System.InvalidOperationException: Camellia · Nanahira - Artificial Intelligence (BreadKnightRG) already has a difficulty with the name of 'Automaton'.
2022-06-26 23:54:01 [error]: at osu.Game.Beatmaps.BeatmapManager.Save(BeatmapInfo beatmapInfo, IBeatmap beatmapContent, ISkin beatmapSkin) in /var/lib/buildkite-agent/builds/debian-gnu-linux-vm-1/ppy/osu/osu.Game/Beatmaps/BeatmapManager.cs:line 307
2022-06-26 23:54:01 [error]: at osu.Game.Screens.Edit.Editor.Save() in /var/lib/buildkite-agent/builds/debian-gnu-linux-vm-1/ppy/osu/osu.Game/Screens/Edit/Editor.cs:line 369
2022-06-26 23:54:01 [verbose]: ⚠️ Camellia · Nanahira - Artificial Intelligence (BreadKnightRG) already has a difficulty with the name of 'Automaton'.
```

Note that this should have also shown up as a notification, did you dismiss is?

**Comment by LargerManZ** [comment]
yes, and I read it so I saved it as something else afterwards. I gave the difficulty a different name. I named it Auto-MAGIC!!.

**Comment by peppy** [comment]
Please attach your `client.db`.

**Comment by LargerManZ** [comment]
I don't have one in my folder, I searched by that file name.

**Comment by peppy** [comment]
`client.realm`, sorry.

**Comment by LargerManZ** [comment]
Github doesn't support that file type how do I go about this

**Reply by peppy** [reply]
You can compress it to a zip file.

**Reply by LargerManZ** [reply]
[client.realm.zip](https://github.com/ppy/osu/files/8990252/client.realm.zip)

**Reply by LargerManZ** [reply]
I shall sleep and be back in a good 8 hours

**Reply by peppy** [reply]
To confirm, you were making a new beatmap from scratch?

I can't find much in your database. I'd suggest you try the whole process again and see if you can reproduce it reliably. If so, a video showing the procedure you're taking to reproduce it would be very helpful.

**Reply by LargerManZ** [reply]
Yes I was making a new beatmap from scratch

**Reply by LargerManZ** [reply]
https://user-images.githubusercontent.com/62466706/176016539-ad9bad4c-37da-4a10-90e9-6e318f638832.mp4

**Comment by LargerManZ** [comment, marked as answer]
replies is answer to my stupidity

**Reply by bdach** [reply]
i've checked the video above but it doesn't tell me anything useful, there are also no accompanying logs. i can assure you the editor saves correctly on my linux installation.

please reupload your `client.realm` file.

**Reply by LargerManZ** [reply]
[client.realm.zip](https://github.com/ppy/osu/files/9004934/client.realm.zip)
[database.log](https://github.com/ppy/osu/files/9004937/database.log)
[network.log](https://github.com/ppy/osu/files/9004938/network.log)
[performance.log](https://github.com/ppy/osu/files/9004939/performance.log)
[runtime.log](https://github.com/ppy/osu/files/9004940/runtime.log)

**Reply by LargerManZ** [reply]
I tried to make another beatmap before sending these, and same results. No maps were created when I saved.

**Reply by bdach** [reply]
i see evidence of at least 3 copies of the map you tried to create:

![image](https://user-images.githubusercontent.com/20418176/176299249-2984abb4-76a3-4d03-b455-35583996d387.png)
![image](https://user-images.githubusercontent.com/20418176/176299647-a740e96d-362a-4ada-9c42-dc3468df966b.png)

to confirm for my own sanity - can you take a screenshot of the following settings item?

![image](https://user-images.githubusercontent.com/20418176/176297582-bfa4e205-4e88-4376-806d-d6dbcc0359f8.png)

**Reply by LargerManZ** [reply]
u right when u right
there is no bug
